MontyTRC89/TombEngine
# Soft intelligent object collision

Makes collision with intelligent objects "soft" in the sense that the player will be pushed away gradually instead of snapping instantly to the edge of its bounding box.

To test:
- Collisions with creatures.
- Creature melee attacks.
- Collision deflection with rotated intelligent objects, i.e. not axis-aligned. Try uninitialised shiva, mummy, etc.
- Ensure collision with non-intelligent objects remains the same.
- Identify any objects flagged "intelligent" as a hack.
